### DISCUSSION TITLE
Is it better to upload files as a PDF or Microsoft word document?

### DISCUSSION THREAD

**Discussion by akim67212** [community/community#142728] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I was just wondering is it better to upload files as a PDF or Microsoft word document? I've noticed that every time I try to upload a new file in a folder of my Repository as a Microsoft word document, when I try to click on the file, it states that they can't show files that are this big and gives me an option to view the document raw. However I do not have this issue if I convert a document as a pdf. The PDF is displayed within my folder without incident. Which is option is more presentable for my repository?

**Comment by LismaxB** [comment, marked as answer]
PDFs are generally more reliable for online viewing, especially for larger files. Word documents can sometimes cause display issues.

**Reply by LismaxB** [reply]
For best display on GitHub, PDFs are generally recommended. They're reliable and easily viewable. Word documents can sometimes have compatibility issues and may not display correctly.

**Reply by akim67212** [reply]
@LismaxB Thank you for answering my question!

**Comment by LismaxB** [comment]
For best display on GitHub, PDFs are generally recommended. They're reliable and easily viewable. Word documents can sometimes have compatibility issues and may not display correctly.

**Comment by MasterInQuestion** [comment]
Plain Text or HTML more preferable.
    For GitHub context, Markdown also.

**Reply by akim67212** [reply]
@MasterInQuestion Thank you!

**Reply by akim67212** [reply]
@MasterInQuestion Can you add images to plain text?

**Reply by LismaxB** [reply]
@akim67212 You can add images in GitHub using Markdown syntax.
- Upload the image file you want into your repo
- Reference the image in your Markdown file:
`![Alt text](path/to/image.jpg)`

#### Unfortunately you can't directly add images to plain text files.
I prefer using Markdown files as it's easier.

**Reply by MasterInQuestion** [reply]
At many times directly embedding images may not be necessary: [1]
    Linking to which should suffice.
    E.g.
    https://github.com/orgs/community/discussions/141073#discussioncomment-11063891
    https://github.com/orgs/community/discussions/63813#discussioncomment-10963537
    https://github.com/orgs/community/discussions/139005#discussioncomment-11046170
    (copy the rendered text and compare)

[ [1]
    At times even undesirable, e.g.
    [https://github.com/orgs/community/discussions/139005#discussioncomment-10889895](https://github.com/orgs/community/discussions/139005#discussioncomment-10889895) ]

    ----

    Easier or not, hard to tell.
    As Markdown has several slightly incompatible syntax flavors:
    That causes significant bother of portability.
    [ Also the general interoperability impairment of alike post-render required schemes. ]

    For markup languages in general:
    https://github.com/MasterInQuestion/Markup

**Reply by akim67212** [reply]
@LismaxB  Can you add images to a Markdown file? And If you can, which is more presentable for a portfolio, a Markdown file or a PDF?

**Reply by MasterInQuestion** [reply]
https://en.wikipedia.org/wiki/Markdown
    https://github.github.com/gfm/
    https://docs.github.com/en/get-started/writing-on-github
    https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

    Probably raw images; merely for presentation.

    Note:
    If presentation is sourced on controllable device:
    Screencasting HTML shall be also viable.

**Reply by LismaxB** [reply]
@akim67212 Yes of course, even when you're replying for this, you can use the markdown format, In github, most of the users uses and prefer markdown as it's ease of use.

#### You can easily add an image to a markdown file,

`![Alt](/add-your-path-or-url-here/image.png)`

**Reply by akim67212** [reply]
@LismaxB Which do you think is more presentable for a portfolio in a repository? A markdown file or a PDF?

**Reply by LismaxB** [reply]
@akim67212 I personally think markdown is better in this  scenario.

**Reply by MasterInQuestion** [reply]
The reason formats like PDF, DOC etc. being disfavored is primarily:
    Unjustifiable complexity and bad general interoperability.
    Plus potential security risks due to some interfaces' design.

**Comment by Amit-Verma5262** [comment]
Uploading files as PDFs is generally the more presentable and compatible choice for a GitHub repository, especially if your goal is for files to display nicely within the repository’s web interface.

**Reply by MasterInQuestion** [reply]
Behavior of handling PDF is unspecified for browsers.
    For screen media, the page assertion of PDF alike also feels much malapropos.

    And there're numerous reports PDF failing to load in Repositories:
    https://github.com/orgs/community/discussions/64419
    https://github.com/orgs/community/discussions/62549

    See also:
    https://github.com/orgs/community/discussions/36947
    https://github.com/orgs/community/discussions/48074
    https://github.com/orgs/community/discussions/129664#discussioncomment-9864889
    https://github.com/orgs/community/discussions/133329
